pkg/identity/cache/allocator.go

```diff
@@ -122,19 +126,26 @@ func Close() {
 	setupMutex.Lock()
 	defer setupMutex.Unlock()
 
-	if IdentityAllocator == nil {
-		log.Panic("Close() called without calling InitIdentityAllocator() first")
+	select {
+	case <-identityAllocatorInitialized:
+		// This means the channel was closed and therefore the IdentityAllocator == nil will never be true
+	default:
+		if IdentityAllocator == nil {
+			log.Panic("Close() called without calling InitIdentityAllocator() first")
+		}
 	}
 
 	IdentityAllocator.Delete()
 	watcher.stop()
 	IdentityAllocator = nil
+	identityAllocatorInitialized = make(chan struct{})
 	localIdentities = nil
 }
 
 // WaitForInitialIdentities waits for the initial set of security identities to
 // have been received and populated into the allocator cache
 func WaitForInitialIdentities() {
+	<-identityAllocatorInitialized
```

Comment:
I assume this is the primary blocker on bootstrap?

Reply:
I didn't get the question

Reply:
Never mind, I see that this is executed in an async go routine as well.